lib/game.rb

```diff
@@ -20,33 +20,45 @@ def toggle_current_player
   end
 
   def make_move
-    @display.display_board(@board.board)
+    @display.display_board(@board.squares)
     @display.show_current_player(@current_player.mark)
     move = @display.ask_for_move
-    until move_valid?(move)
-      @display.invalid_move_message(move)
+    until @board.move_valid?(move)
+      @display.show_invalid_move_message(move)
       move = @display.ask_for_move
     end
     @board.player_make_move(@current_player.mark, move.to_i)
   end
 
-  public
+  def winning_player_exists?
+    @board.has_winning_combination?(@player1.mark) || @board.has_winning_combination?(@player2.mark)
+  end
 
-  def moves_remaining?
-    @board.moves_remaining > 0
+  def winning_player
+    if @board.has_winning_combination?(@player1.mark)
+      @player1.mark
+    else
+      @player2.mark
+    end
   end
 
+  public
+
   def play_move
     make_move
     toggle_current_player
   end
 
-  def can_continue_playing?
-    moves_remaining?
+  def is_over?
+    !@board.moves_remaining? || winning_player_exists?
```

Comment:
I would avoid the negation where possible as the `||` bring some complexity. If you had a `board.complete?` (which can just be the opposite of `moves_remaining?` then this would read more naturally as:
```
@board.complete? || winning_player_exists?
```

Reply:
Thank you. Changing it to `@board.complete?` makes it much easier to read than !@board.moves_remaining?